flashcard-quiz/src/main.rs

```diff
@@ -0,0 +1,246 @@
+    Edit {
+        index: usize,
+        question: Option<String>,
+        answer: Option<String>,
+    },
+    /// Remove an existing flashcard
+    Remove { index: usize },
+    /// Show all flashcards
+    Show,
+    /// Start the quiz
+    Quiz {
+        #[clap(short, long, default_value = "5")]
+        num_questions: usize,
+    },
+}
+
+#[derive(Default, serde::Serialize, serde::Deserialize)]
+struct Quiz {
+    cards: Vec<Card>,
+}
+
+#[derive(serde::Serialize, serde::Deserialize)]
+struct Card {
+    question: String,
+    answer: String,
+}
+
+fn main() -> anyhow::Result<()> {
+    let args = Args::parse();
+
+    let mut quiz = load_or_create_quiz()?;
+
+    match args.subcommand {
+        Subcommand::Add { question, answer } => add_card(&mut quiz, &question, &answer),
+        Subcommand::Edit {
+            index,
+            question,
+            answer,
+        } => edit_card(&mut quiz, index - 1, question.as_deref(), answer.as_deref())?,
+        Subcommand::Remove { index } => remove_card(&mut quiz, index - 1)?,
+        Subcommand::Show => show_cards(&quiz),
+        Subcommand::Quiz { num_questions } => start_quiz(&quiz, num_questions)?,
+    }
+
+    save_quiz(&quiz)?;
+
+    Ok(())
+}
+
+fn load_or_create_quiz() -> anyhow::Result<Quiz> {
+    let path = Path::new("quiz.json");
+    if path.exists() {
+        let file = File::open(path)?;
+        let quiz = serde_json::from_reader(file)?;
+        Ok(quiz)
+    } else {
+        let quiz = Quiz::default();
+        Ok(quiz)
+    }
+}
+
+fn save_quiz(quiz: &Quiz) -> anyhow::Result<()> {
+    let file = File::create("quiz.json")?;
+    serde_json::to_writer(file, quiz)?;
+    Ok(())
+}
+
+fn add_card(quiz: &mut Quiz, question: &str, answer: &str) {
+    quiz.cards.push(Card {
+        question: question.to_string(),
+        answer: answer.to_string(),
+    });
+
+    println!(
+        "Added flashcard #{} with question: '{}' and answer: '{}'",
+        quiz.cards.len(),
+        question,
+        answer
+    );
+}
+
+fn edit_card(
+    quiz: &mut Quiz,
+    index: usize,
+    question: Option<&str>,
+    answer: Option<&str>,
+) -> anyhow::Result<()> {
+    let card = quiz
+        .cards
+        .get_mut(index)
+        .ok_or_else(|| anyhow::anyhow!("invalid index: {index}"))?;
+
+    if let Some(question) = question {
+        card.question = question.to_string();
+    }
+
+    if let Some(answer) = answer {
+        card.answer = answer.to_string();
+    }
+
+    println!(
+        "Edited flashcard #{} with question: '{}' and answer: '{}'",
+        index + 1,
+        card.question,
+        card.answer
+    );
+
+    Ok(())
+}
+
+fn remove_card(quiz: &mut Quiz, index: usize) -> anyhow::Result<()> {
+    if index >= quiz.cards.len() {
+        return Err(anyhow::anyhow!("invalid index: {index}"));
+    }
+
+    let card = quiz.cards.remove(index);
+
+    println!(
+        "Removed flashcard #{} with question: '{}' and answer: '{}'",
+        index + 1,
```

Comment:
<details> 
  <summary>Corey's Third Bug Guess [but its really the same lol]</summary>
  This actually doesn't need to be adjusted since its straight from the User

So we can just use `index` here
</details>